ipn/localapi/localapi.go

```diff
@@ -915,6 +915,15 @@ func (h *Handler) serveServeConfig(w http.ResponseWriter, r *http.Request) {
 			writeErrorJSON(w, fmt.Errorf("decoding config: %w", err))
 			return
 		}
+
+		// require a local admin when setting a path handler
+		// TODO: roll-up this Windows-specific check into either PermitWrite
+		// or a global admin escalation check.
+		if runtime.GOOS == "windows" && configIn.HasPathHandler() && !h.CallerIsLocalAdmin {
```

Comment:
maybe pull this out into a policy func that returns a bool, then it's table testable and a pure func.

inputs: goos string, configIn, localapi.Handler (h)